### PR TITLE
fix(menu): fixed a bug where cascading child menus would overlap when clipped

### LINKED ISSUE #257: [menu] cascading menus are not positioning themselves properly
**Describe the bug:**
When using a cascading menu with 2+ levels of nesting on the far right of the screen (where opening a child menu would not have enough room), the child menus are stacking on top of the parent menu.

This is caused by the child menu placement being hardcoded here:
https://github.com/tyler-technologies-oss/forge/blob/04741554822b7b4b5831d5146cf1ed4f4ed225a0/src/lib/menu/menu-foundation.ts#L456

We need the positioning logic to automatically choose the opposite side on the same axis instead (which the `<forge-popup>` supports already).

**To Reproduce:**
1. Create a multi-level cascading menu that is positioned to the far right of the screen so there isn't enough room to expand open a child cascading menu.
2. Hover over the child menu
3. Notice how the child menu stacks on top of its parent menu instead of the opposite side

**Expected behavior:**
The child menu should instead automatically attempt to position itself on the opposite side of the same axis it originally opened on (left side in this case first).

**Screenshots:**
As seen below, the cascading child menus have no room to the right to open themselves and are stacking on top of the parent menu. This causes issues when trying to move the mouse to the proper menu item.
image

**Please complete the following information:**
 - Forge version: ^2.0.0
 - I have searched existing issues before creating this report? Y
 - Browser: All
 - Platform: All
 - OS: All

**Additional context:**
The `<forge-popup>` positioning logic already supports this functionality, we should be able to tap into that.

### PR DESCRIPTION
**Pull request by DRiFTy17**
## PR Checklist

Please check if your PR fulfills the following requirements:

- Tests for the changes have been added: N
- Docs have been added / updated: Y
- Does this PR introduce a breaking change? N
- I have linked any related GitHub issues to be closed when this PR is merged? Y

## Describe the new behavior?
Cascading menus will now use a specific fallback placement strategy of horizontal-only when opening.

image

## Additional information
Fixes #257

### CODE REVIEW

File: src/dev/src/partials/header.ejs
Comment (DRiFTy17) [inline src/dev/src/partials/header.ejs:1]: Unrelated change to enhance the demo page with a home link for easier navigation.

File: src/lib/autocomplete/autocomplete-constants.ts [hunk 1]
```diff
@@ -76,6 +76,6 @@ export interface IAutocompletePopupConfiguration {
   scrollEndListener: () => void;
 }
 
-export interface IAutocompleteSelectEventData {
-  value: any;
+export interface IAutocompleteSelectEventData<T = any> {
+  value: T;
 }
```
Comment (DRiFTy17) [inline src/lib/autocomplete/autocomplete-constants.ts:81]: Included this unrelated change here to enhance the typings.

**Review by samrichardsontylertech — DISMISSED** [review]
